cni/pkg/repair/repair.go

```diff
@@ -102,13 +87,16 @@ func (bpr BrokenPodReconciler) LabelBrokenPods() (err error) {
 
 func (bpr BrokenPodReconciler) labelBrokenPod(pod v1.Pod) (err error) {
 	// Added for safety, to make sure no healthy pods get labeled.
+	m := podsRepaired.With(typeLabel.Value(labelType))
 	if !bpr.detectPod(pod) {
+		m.With(resultLabel.Value(resultSkip)).Increment()
 		return
 	}
 	log.Infof("Pod detected as broken, adding label: %s/%s", pod.Namespace, pod.Name)
 
 	labels := pod.GetLabels()
 	if _, ok := labels[bpr.Options.PodLabelKey]; ok {
+		m.With(resultLabel.Value(resultSkip)).Increment()
```

Comment:
I think adding skip will mean someone doing an alert on `istio_cni_repair_pods_repaired_total > 0` will not trigger? I think I read somewhere having failures be a separate metric was a best practice but cannot find it now so I am not sure.. WDYT?

Reply:
Alert should be based on `istio_cni_repair_pods_repaired_total{result=fail} > 0`. I don't think a separated metric is necessary. Like we don't have a `istio_request_errors_total` for the standard metric.

Reply:
Their existing alerts will be on `istio_cni_repair_pods_repaired_total > 0` and would break on upgrade though?

Reply:
In a good cluster you should have no repairs - a repair happening means we hit the race condition

Reply:
And this is not in 1.9, so not so much of a concern? We can cherry-pick this change into 1.10.

Reply:
I guess maybe you wouldn't have an alert though

Reply:
Feel free to remove the hold

Reply:
Yeah I doubt anyone would alert on race conditions if they know there will be such a race. If someone who solves the race in other way still want to alert on this, they can still do `istio_cni_repair_pods_repaired_total > 0` since even `skip` is problematic.